includes/multi-currency/Notes/NoteMultiCurrencyAvailable.php

```diff
@@ -28,10 +29,22 @@ class NoteMultiCurrencyAvailable {
 	// TODO: Proper url needed for setup process.
 	const NOTE_SETUP_URL = 'admin.php?page=wc-admin&path=/payments/multi-currency-setup';
 
+	/**
+	 * The account service instance.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private static $account;
+
 	/**
 	 * Get the note.
 	 */
 	public static function get_note() {
+		// Don't show the note if the merchant didn't create a WCPay account yet.
+		if ( ! self::$account->is_stripe_connected() ) {
```

Comment:
Moving this condition to `possibly_add_note` makes more sense and ensure that this method is a pure getter without any side effects. 

The same pattern is being used in `WC_Payments_Notes_Loan_Approved::possibly_add_note` - [src](https://github.com/Automattic/woocommerce-payments/blob/1d5ce3ce89c9a2513c036c23fd5a5c4fa6d266c4/includes/notes/class-wc-payments-notes-loan-approved.php#L88-L88).

Also, there is a chance that `self::$account` is not set yet at this point, so checking it too makes code a bit more defensive. 

```suggestion
		if ( is_null ( self::$account ) || ! self::$account->is_stripe_connected() ) {
```

Reply:
Agreed that it would be best to move the check into possibly_add_note, done.
Also added an additional check on the account as you suggested to be sure it won't break.